lib/algolia/search_index.rb

```diff
@@ -987,7 +987,20 @@ def get_settings(opts = {})
       # @return [IndexingResponse]
       #
       def set_settings(settings, opts = {})
-        response = @transporter.write(:PUT, path_encode('/1/indexes/%s/settings', @name), settings, opts)
+        forward_to_replicas  = false
+        request_options      = symbolize_hash(opts)
+
+        if request_options[:forwardToReplicas]
+          forward_to_replicas = true
+          request_options.delete(:forwardToReplicas)
+        end
```

Comment:
The [`Hash#delete`](https://apidock.com/ruby/Hash/delete) method returns the value of the corresponding key, or `nil` if not defined. We can use that behavior to shorten the code a lot:
```suggestion
        request_options      = symbolize_hash(opts)
        forward_to_replicas  = request_options.delete(:forwardToReplicas) || false
```